Dockerfile.ci.s390x

```diff
@@ -0,0 +1,297 @@
+FROM ubuntu
+MAINTAINER Alejandro Martinez Ruiz <amr@redhat.com>
+
+ARG CIRCLECI_TOOLS="git openssh-client tar gzip ca-certificates locales locales-all"
+# set localtime to UTC and install basic build tools
+RUN (ln -s /usr/share/zoneinfo/UTC /etc/localtime 2> /dev/null || true) \
+    && apt update -y \
+    && apt install -y sudo git make gcc autoconf automake libtool gettext file \
+       ${CIRCLECI_TOOLS} \
+    && apt clean all \
+    && localedef -c -f UTF-8 -i en_US en_US.UTF-8
+
+ARG USER_NAME=ruby
+RUN addgroup wheel
+RUN useradd -m -d "/home/${USER_NAME}" -l -G wheel ${USER_NAME} \
+    && echo "${USER_NAME} ALL=(ALL:ALL) NOPASSWD: ALL" > \
+    "/etc/sudoers.d/${USER_NAME}" \
+    && chmod 0440 "/etc/sudoers.d/${USER_NAME}"
+
+USER "${USER_NAME}"
+
+# PostgreSQL installation
+ARG POSTGRES_BUILD_DEPS="bison flex zlib1g-dev libreadline-dev libssl-dev"
+RUN test "x${POSTGRES_BUILD_DEPS}" = "x" \
+    || sudo apt install -y ${POSTGRES_BUILD_DEPS}
+
+ARG POSTGRES_VERSION=REL_10_17
+RUN curl -sSLf "https://github.com/postgres/postgres/archive/${POSTGRES_VERSION}.tar.gz" \
+    | tar -C "/home/${USER_NAME}" -xz
+
+ARG POSTGRES_PREFIX=/usr/local
+ARG POSTGRES_DATA_PREFIX=${POSTGRES_PREFIX}/pgsql
+ARG POSTGRES_CONFIGURE_OPTIONS
+ARG POSTGRES_MAKE_OPTIONS
+RUN cd ~/postgres-"${POSTGRES_VERSION}" \
+    && ./configure --prefix="${POSTGRES_PREFIX}" --includedir=/usr/local/include \
+       --with-openssl ${POSTGRES_CONFIGURE_OPTIONS} \
+    && make ${POSTGRES_MAKE_OPTIONS}
+RUN cd ~/postgres-"${POSTGRES_VERSION}" \
+    && sudo make install \
+    && rm -rf ~/postgres-"${POSTGRES_VERSION}" \
+    # the test tooling expects the pg_ctl binary in this directory
+    && sudo mkdir -p /usr/lib/postgresql/9.5/bin \
+    && sudo ln -sf "${POSTGRES_PREFIX}/bin/pg_ctl" /usr/lib/postgresql/9.5/bin/pg_ctl \
+    && sudo useradd postgres \
+    && sudo mkdir -p "${POSTGRES_DATA_PREFIX}/data" \
+    && sudo chown -R postgres: "${POSTGRES_DATA_PREFIX}" \
+    && sudo chmod -R go-rwx "${POSTGRES_DATA_PREFIX}"
+
+RUN sudo runuser -l postgres -c \
+    "${POSTGRES_PREFIX}/bin/initdb --pgdata='${POSTGRES_DATA_PREFIX}/data' --auth='trust'" < /dev/null \
+    && sudo runuser -l postgres -c \
+    "${POSTGRES_PREFIX}/bin/postgres -D '${POSTGRES_DATA_PREFIX}/data'" \
+      > /tmp/postgres.log 2>&1 & sleep 5 \
+    && sudo runuser -l postgres -c "${POSTGRES_PREFIX}/bin/createdb test" \
+    && sudo runuser -l postgres -c "${POSTGRES_PREFIX}/bin/psql test"
+
+# Twemproxy installation
+ARG TWEMPROXY_VERSION=0.5.0
+RUN curl -sSLf \
+    "https://github.com/twitter/twemproxy/releases/download/0.5.0/twemproxy-${TWEMPROXY_VERSION}.tar.gz" \
+    | tar -C "/home/${USER_NAME}" -xz
```

Comment:
I know this has been carried over from `Dockerfile.ci`, but it makes little sense to be able to select the `TWEMPROXY_VERSION` via an argument only to encode the version in the path anyway. That change slipped somehow into the other file, but should be fixed, either by reusing `TWEMPROXY_VERSION` in its stead, or by finding another download link.